src/main/java/ca/corefacility/bioinformatics/irida/ria/config/thymeleaf/webpacker/util/WebpackerManifestParser.java

```diff
@@ -0,0 +1,125 @@
+package ca.corefacility.bioinformatics.irida.ria.config.thymeleaf.webpacker.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.ResourceUtils;
+
+import ca.corefacility.bioinformatics.irida.processing.FileProcessorException;
+
+/**
+ * Responsible for parsing the webpack manifest file and passing along the chunks for
+ * js, css, and html resources.  During development the manifest file is checked to ensure
+ * it has not been updated during a build, this will not happen in production since the
+ * manifest file will never change.
+ */
+public class WebpackerManifestParser {
+	private static final Logger logger = LoggerFactory.getLogger(WebpackerManifestParser.class);
+	private static Map<String, Map<String, Map<String, List<String>>>> entryMap;
```

Comment:
That is a good question, I will play with that for a bit and try to figure something out.  It is super ugly and hard to maintain, I definitely agree with this.

Reply:
Check out cb655c76b04d9e191d4f24a879607435fe51b2c1 and let me know what you think

Reply:
Updated in d6ef6cd

Reply:
That's definitely better, still not the best that you have the `Map<String, Map<String, Map<String, List<String>>>>` going on when `manifest` is being read from the `ObjectMapper`.  I'll think a bit on whether there's a nicer way to deserialize that, otherwise at least the classes should be documented.  Also probably moved into their own files.